### PR TITLE
Udev remove unit test

### PR DESCRIPTION
**Pull request by tasleson**
Automated unit test which reproduces
https://github.com/stratis-storage/stratisd/issues/1197

This will require the following changes on the CI test nodes:

* nbd kernel module is loaded
* `/usr/lib/udev/rules.d/60-block.rules` needs `|nbd*`  see: https://github.com/systemd/systemd/pull/11676
* nbd and nbdkit packages installed `dnf install nbd nbdkit nbdkit-basic-plugins`

### CODE REVIEW

**Comment by tasleson** [comment]
Rebase

**Comment by mulkieran** [comment]
Note: Behavior is what is to be expected if running on most recent devicemapper version: https://stratis-jenkins.rhev-ci-vms.eng.rdu2.redhat.com/job/stratisd/2696/console, since stratisd will fail on an assertion. If running off master devicemapper, though, stratisd would not hit the assertion, since now the function returns an error in that case.

File: tests/client-dbus/tests/udev/remove/test_remove.py [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+        """
+        super(UdevRemove, self).setUp()
+
+        result = flight_check()
+        if result:
+            raise unittest.SkipTest(result)
+
+        self._blk_mgr = NbdDevices()
+
+    def test_single_device_in_pool(self):
```
Comment (mulkieran) [inline tests/client-dbus/tests/udev/remove/test_remove.py]: Please decorate test with ```@unittest.expectedFailure``` and run it again.

**Comment by tasleson** [comment]
@mulkieran 

> The test itself also has an error, because stratisd is no longer running. Let's assume that stratisd should always be running at the end of a successful test, for now.

The test is ensuring that the daemon is running when we are told to stop, so I'm confused about this statement?

**Comment by tasleson** [comment]
> Is there any way to get rid of newstyle/oldstyle nbd warnings?

Done

**Comment by mulkieran** [comment]
Seems like Travis failure is just due to a formatting error.

File: tests/client-dbus/tests/udev/remove/test_remove.py [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+
+    def setUp(self):
+        """
+        Common needed things
+        """
+        super(UdevRemove, self).setUp()
+
+        result = flight_check()
+        if result:
+            raise unittest.SkipTest(result)
```
Comment (mulkieran) [inline tests/client-dbus/tests/udev/remove/test_remove.py]: Try just calling ```self.skip()``` here.

Comment (tasleson) [reply tests/client-dbus/tests/udev/remove/test_remove.py]: Done with https://github.com/stratis-storage/stratisd/pull/1438/commits/b9b3a738fa9b89bab27f1766e4d7f5741df35e24

**Comment by tasleson** [comment]
@mulkieran @trgill Rebased, re-worked a couple things and made suggested edits.

**Comment by tasleson** [comment]
@trgill @mulkieran Updated and squashed a bit.  Local testing is showing this to be very stable.

**Comment by trgill** [comment]
@bgurney-rh @drckeefe @mulkieran please discuss in standup?

**Comment by mulkieran** [comment]
We should consider this blocked by #1502, since none of the tests being changed are being run at all.

**Comment by mulkieran** [comment]
No longer blocked by #1502.

**Comment by mulkieran** [comment]
@tasleson Please rebase.

**Comment by tasleson** [comment]
> @tasleson Please rebase.
@mulkieran Done

**Comment by tasleson** [comment]
> Please remove requirement that STRATIS_DESTRUCTIVE_TEST be set on new remove test.

Yes, forgot I had that in there, thanks.  Added removal.

**Comment by tasleson** [comment]
Re-based, conflicts resolved, commits squashed some more.

**Comment by tasleson** [comment]
I remembered that some testing was done with Stratis which yielded similar test results as this one, but it requires less functionality for the unit test.  Basically you can do a less invasive remove by doing

```
echo offline > /sys/block/<devname>/device/state
```

You can restore operation by doing:
```
echo running > /sys/block/<devname>/device/state
```
Note: No udev events are created by this 

Thus it would be easier to add an existing test to the existing udev tests which creates a pool with one or more loop back devices and then does the `offline` operation on one or more of them.  Our expectation is that the stratisd process continues to run and you can query the pool(s) and get the expected number returned.  Today I believe we don't crash the service, but when we retrieve all the managed objects we get an error as dm cannot talk to the device to retrieve information needed to fill out the properties.

**Comment by tasleson** [comment]
Wrote up issue to add this unit test using easier method than this: https://github.com/stratis-storage/stratisd/issues/1512

**Pull request closed without merging**